### PR TITLE
Refactoring is in doctests for slow downloads

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Changes the logic introduced in https://github.com/Lightning-AI/metrics/pull/1505 to an env variable instead.
Additionally, add skipping to rouge metric if download is not available as it sometimes silently halts (and there timeout the tests).

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/torchmetrics/utilities/checks.py [hunk 1]
```diff
@@ -758,14 +758,8 @@ def is_overridden(method_name: str, instance: object, parent: object) -> bool:
 
 
 def _in_doctest() -> bool:
-    """Determine if script running in doctest context."""
```
Comment (Borda) [inline src/torchmetrics/utilities/checks.py:761]: Yeah, was testing this locally and didn't work much, so env var looks better...

Comment (Borda) [reply src/torchmetrics/utilities/checks.py:761]: But then we don't need to have it as function?

Comment (SkafteNicki) [reply src/torchmetrics/utilities/checks.py:761]: you are right, function removed :)

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
Shall we extend timeout for the one ci check?

File: tests/unittests/text/test_rouge.py [hunk 2]
```diff
@@ -118,6 +118,7 @@ def _compute_rouge_score(
 class TestROUGEScore(TextTester):
     @pytest.mark.parametrize("ddp", [False, True])
     @pytest.mark.parametrize("dist_sync_on_step", [False, True])
+    @skip_on_connection_issues(reason="could not download nltl relevant data")
```
Comment (stancld) [inline tests/unittests/text/test_rouge.py:121]:
```suggestion
    @skip_on_connection_issues(reason="could not download nltk relevant data")
```

File: tests/unittests/text/test_rouge.py [hunk 3]
```diff
@@ -137,6 +138,7 @@ def test_rouge_score_class(
             key=pl_rouge_metric_key,
         )
 
+    @skip_on_connection_issues(reason="could not download nltl relevant data")
```
Comment (stancld) [inline tests/unittests/text/test_rouge.py:141]:
```suggestion
    @skip_on_connection_issues(reason="could not download nltk relevant data")
```

File: tests/unittests/text/test_rouge.py [hunk 4]
```diff
@@ -197,6 +199,7 @@ def test_rouge_metric_wrong_key_value_error():
         "rougeLsum_fmeasure",
     ],
 )
+@skip_on_connection_issues(reason="could not download nltl relevant data")
```
Comment (stancld) [inline tests/unittests/text/test_rouge.py:202]:
```suggestion
@skip_on_connection_issues(reason="could not download nltk relevant data")
```

File: tests/unittests/text/test_rouge.py [hunk 5]
```diff
@@ -235,6 +238,7 @@ def test_rouge_metric_normalizer_tokenizer(pl_rouge_metric_key):
     ],
 )
 @pytest.mark.parametrize("use_stemmer", [False, True])
+@skip_on_connection_issues(reason="could not download nltl relevant data")
```
Comment (stancld) [inline tests/unittests/text/test_rouge.py:241]:
```suggestion
@skip_on_connection_issues(reason="could not download nltk relevant data")
```

**Review by stancld — APPROVED** [review]

**Pull request merged**